### PR TITLE
Some missing fixes for the Java 8 migration 

### PR DESCRIPTION
**Pull request by iemejia**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -93,7 +93,7 @@
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <external.build.directory>${project.basedir}/target</external.build.directory>
-    <beam.javadoc_opts />
+    <beam.javadoc_opts>-Xdoclint:all -Xdoclint:-missing</beam.javadoc_opts>
```
Comment (kennknowles) [inline pom.xml]: Eventually, we want `missing` back, yes? Or I guess checkstyle also enforces our choice of where they should be?

**Comment by jbonofre** [comment]
By the way, can you squash the commits ? Thanks.

**Comment by kennknowles** [comment]
Commit history looks meaningful, so merging as-is.

**Comment by kennknowles** [comment]
Oh, sorry! Moved too fast and didn't see the comment that happened the same time I was merging.

**Comment by jbonofre** [comment]
No worries, thanks ;)

**Pull request merged**